python-sdk/tests/databases/test_all_databases.py

```diff
@@ -60,3 +61,42 @@ def test_export_table_to_pandas_dataframe(
         ]
     )
     assert df.rename(columns=str.lower).equals(expected)
+
+
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {
+            "database": Database.BIGQUERY,
+            "table": Table(metadata=Metadata(schema=SCHEMA)),
+        },
+        {
+            "database": Database.POSTGRES,
+            "table": Table(metadata=Metadata(schema=SCHEMA.lower())),
+        },
+        {
+            "database": Database.SNOWFLAKE,
+            "table": Table(metadata=Metadata(schema=SCHEMA)),
+        },
+        {"database": Database.SQLITE, "table": Table()},
+    ],
+    indirect=True,
+    ids=["bigquery", "postgres", "snowflake", "sqlite"],
+)
+def test_export_to_dataframe_via_byte_stream_is_called_for_postgres(
+    database_table_fixture,
+):
+    """Test export_to_dataframe_via_byte_stream() is called in case the db is postgres."""
+    database, _ = database_table_fixture
+    file = File(str(pathlib.Path(CWD.parent, "data/sample.csv")))
+
+    with mock.patch(
+        "astro.files.base.File.export_to_dataframe"
+    ) as export_to_dataframe, mock.patch(
+        "astro.files.base.File.export_to_dataframe_via_byte_stream"
+    ) as export_to_dataframe_via_byte_stream:
```

Comment:
Why not have these as decorators above the test function?

Reply:
readability I suppose, the following looks more cleaner that context manager:

```python
@mock.path("astro.files.base.File.export_to_dataframe")
@mock.patch("astro.files.base.File.export_to_dataframe_via_byte_stream")
```

Reply:
@kaxil @dimberman Done.